src/data/set/pointwise/iterate.lean

```diff
@@ -0,0 +1,39 @@
+/-
+Copyright (c) 2022 Oliver Nash. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Oliver Nash
+-/
+import data.set.pointwise.basic
+import algebra.hom.iterate
+import dynamics.fixed_points.basic
+
+/-!
+# Results about pointwise operations on sets with iteration.
+-/
+
+open_locale pointwise
+open set function
+
+/-- Let `n : ℤ` and `s` a subset of a commutative group `G` that is invariant under preimage for
+the map `x ↦ x^n`. Then `s` is invariant under the pointwise action of the subgroup of elements
+`g : G` such that `g^(n^j) = 1` for some `j : ℕ`. (This subgroup is called the Prüfer subgroup when
+ `G` is the circle and `n` is prime.) -/
```

Comment:
```suggestion
 `G` is the `circle` and `n` is prime.) -/
```
assuming that's the right circle.

Reply:
I actually deliberately avoided Leanifying the references to the "circle" in this parenthetical remark because I'm making an informal point. We do not have a formalisation of Prufer groups and I'm trying to refer to them in the abstract (informally) here rather than an implementation of them.

Secondly, though we have both `circle` and `add_circle` we do not have them connected via `to_additive`, and I don't think it would be so easy to do this so it doesn't feel quite right to link to them as suggested.

Of course I've no strong feelings, I will make the changes you suggest if you're not persuaded by these remarks.

Reply:
Even if this is an abstract sense, I think it's still nice to remind the reader that mathlib does have `circle` elsewhere. Happy for you to do whichever; just make sure to fix the grammar / addictiveness of the others comment below

Reply:
Done!